P0009/P0009.bs

```diff
@@ -607,36 +607,42 @@ Description
 ===========
 
 The proposed polymorphic multidimensional array reference (`mdspan`)
-defines types and functions for mapping indices from the **domain**, a
-**multidimensional index space**, to the **codomain**, elements of a
-contiguous span of objects. A multidimensional index space is defined as
-the Cartesian product of integer extents, `[0..N0) * [0..N1) * [0..N2) `....
+defines types and functions for mapping multidimensional indices
+in its **domain**, a
+multidimensional index space, to the `mdspan`'s **codomain**,
+elements of a contiguous span of objects.
+A **multidimensional index space** of **rank** <math>r</math>
+is the Cartesian product 
+<math>[0, N<sub>0</sub>) &#10799; [0, N<sub>1</sub>) &#10799; ... &#10799; [0, N<sub>r-1</sub>)</math>
+of half-open integer intervals.
+A **multidimensional index** or **multi-index**
+is a member of a multidimensional index space.
 An `mdspan` has two policies: the **layout mapping**
 and the **accessor**. The layout mapping specifies the formula, and
 properties of the formula, for mapping a multi-index from the domain to
 an element in the codomain. The accessor is an extension point that
-allows modification of how elements are accessed. For example, the
-Accessors paper (P0367) proposed a rich set of potential access
-properties.
+allows modification of how elements are accessed. For example,
+\[P0367](http://wg21.link/p0367)
+proposed a rich set of potential access properties.
 
-**A multidimensional array is not an array-of-array-of-array-of-array...**
+**A multidimensional array is not an array-of-array-of-array-of...**
 
 The multidimensional array abstraction has been fundamental to numerical
 computations for over five decades. However, the C/C++ language provides
 only a one-dimensional array abstraction which can be composed into
-array-of-array-of-array... types. While such types have some similarity
+array-of-array-of-array-of... types. While such types have some similarity
 to multidimensional arrays, they do not provide adequate multidimensional
 array functionality of this proposal. Two critical functionality
 differences are (1) multiple dynamic extents and (2) polymorphic mapping
 of multi-indices to element objects.
 
 **Optimized Implementation of Layout Mapping**
 
-The layout mapping of a multi-index is intended to be an O(1) `constexpr`
-operation that is trivially inlined and optimized. Note that Fortran
-compilers' optimizations include loop invariant code motion, including
-partial evaluation of multi-index layout mappings when indices are
-loop-invariant.
+We intend the layout mapping of a multi-index to be a constant-time
+`constexpr` operation that is trivially inlined and optimized.
+Compiler vendors may apply optimizations such as loop-invariant code
+motion, including partial evaluation of multi-index layout mappings
+when indices are loop invariant.
```

Comment:
(minor note) "loop-invariant" was correct in the first place, since invariant is an adjective rather than a verb here, and "loop" doesn't have the traditional form of an adverb.  (Just a grammar nit-pick :-P)

Reply:
Fixed in the comments I just pushed.